tests/lib/hardware/hardware_base.py

```diff
@@ -79,6 +80,17 @@ def generate_keys(self):
             "%s%s_key" % (config.CLUSTER_PREFIX, self.hardware_uuid)
         self.pubkey = "%s %s" % (key.get_name(), key.get_base64())
 
+    def ssh_agent(self):
+        res = subprocess.run(['ssh-agent'], capture_output=True)
+        self.SSH_AUTH_SOCK = res.stdout.decode(
+            'utf-8').split(';')[0].split('=')[1]
+        self.SSH_AGENT_PID = res.stdout.decode(
+            'utf-8').split(';')[2].split('=')[1]
+        os.environ['SSH_AUTH_SOCK'] = self.SSH_AUTH_SOCK
+        os.environ['SSH_AGENT_PID'] = self.SSH_AGENT_PID
```

Comment:
I'm not sure how this make things clearer ? SSH_AUTH_SOCK is a standard env variables why pass it to every call ?

Reply:
It makes the code more readable in my opinion. eg:
 * If a developer is looking at how Ansible is called in `ansible_helper.py`, it would be reasonable for them to expect to be able to set the SSH_* env vars. However, they might miss that they are actually being overwritten in this other file.
 * Additionally it makes it clearer in the `ansible_helper.py` that an agent is being configured with a variable (rather than relying on what might be set by the system).

Reply:
Actually I'm no setting the agent for ansible at all.
It is used for skuba only so it has been moved to Caasp()